var/spack/repos/builtin/packages/boost/package.py

```diff
@@ -593,21 +616,32 @@ def install(self, spec, prefix):
 
         # b2 used to be called bjam, before 1.47 (sigh)
         b2name = './b2' if spec.satisfies('@1.47:') else './bjam'
+        if self.spec.satisfies('platform=windows'):
+            b2name = 'b2.exe' if spec.satisfies('@1.47:') else 'bjam.exe'
 
         b2 = Executable(b2name)
         jobs = make_jobs
+        path_to_config = ''
+        if not spec.satisfies('platform=windows'):
+            path_to_config = '--user-config=%s' % os.path.join(
+                self.stage.source_path, 'user-config.jam')
         # in 1.59 max jobs became dynamic
         if jobs > 64 and spec.satisfies('@:1.58'):
             jobs = 64
 
-        b2_options = [
-            '-j', '%s' % jobs,
-            '--user-config=%s' % os.path.join(
-                self.stage.source_path, 'user-config.jam')
-        ]
+        b2_options = ['-j', '%s' % jobs]
+        if not spec.satisfies('platform=windows'):
+            b2_options.append(path_to_config)
 
         threading_opts = self.determine_b2_options(spec, b2_options)
 
+        # Windows build just wants a b2 call with no args
+        if spec.satisfies('platform=windows'):
+            b2_options.clear()
```

Comment:
Between this and the conditional application of `--user-config` (i.e. that is skipped entirely on Windows` I think it would be more-straightforward to reorganize this function (or more precisely everything after the `if not with_libs` block) into an if/else based on `spec.satisfies(platform=windows)`